sources/packages/web/src/components/aest/students/bypassedRestrictions/HistoryBypassedRestrictions.vue

```diff
@@ -107,6 +161,40 @@ export default defineComponent({
       return !isBypassActive || !isRestrictionActive ? "secondary" : "primary";
     };
 
+    const openBypassRestrictionModal = async () => {
+      await bypassRestrictionModal.value.showModal();
+    };
+    const openRemoveBypassModal = async (
+      applicationRestrictionBypassId: number,
+    ) => {
+      selectedApplicationRestrictionBypassId.value =
+        applicationRestrictionBypassId;
+      await removeBypassModal.value.showModal();
+    };
+
+    const openBypassViewDetailsModal = async (
+      applicationRestrictionBypassId: number,
+    ) => {
+      selectedApplicationRestrictionBypassId.value =
+        applicationRestrictionBypassId;
+      await bypassRestrictionModal.value.showModal();
+      selectedApplicationRestrictionBypassId.value = 0;
+    };
+
+    const loadBypassedRestrictionHistory = async () => {
+      bypassedRestrictions.value =
+        await ApplicationRestrictionBypassService.shared.getApplicationRestrictionBypassesHistory(
+          props.applicationId,
+        );
+    };
+
+    const reloadRestrictionBypassedHistory = () => {
+      loadBypassedRestrictionHistory();
+      bypassRestrictionModalKey.value += 1;
+      removeBypassModalKey.value += 1;
+      selectedApplicationRestrictionBypassId.value = 0;
+    };
```

Comment:
We have other approaches to avoid this type of code, as far as I know.
The below is one of the most recent modals, please take a look if it helps as a reference.
https://github.com/bcgov/SIMS/blob/595f6374c04595f0254f84f6f861e10c4ea35e45/sources/packages/web/src/components/aest/students/modals/EditStudentProfileModal.vue